interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,315 @@
-            HDCP_UNAVAILABLE,
-            HDCP_14,
-            HDCP_20,
-            HDCP_21,
-            HDCP_22
+            HDCP_UNAVAILABLE /* @text: unavailable */,
+            HDCP_14 /* @text: 1.4 */,
+            HDCP_20 /* @text: 2.0 */,
+            HDCP_21 /* @text: 2.1 */,
+            HDCP_22 /* @text: 2.2 */
+        };
+
+        struct VideoOutputCaps {
+            IDeviceVideoCapabilities::VideoOutput videoOutput /* @brief Video Output support */;
+            IDeviceVideoCapabilities::CopyProtection hdcp /* @brief Video Output support */;
+            IDeviceVideoCapabilities::ScreenResolution outputResolutions /* @bitmask @brief Supported resolutions */;
+            IDeviceVideoCapabilities::ScreenResolution defaultResolution /* @brief Default resolution */;
+        };
+
+        using IVideoOutputCapsIterator = RPC::IIteratorType<VideoOutputCaps, ID_DEVICE_CAPABILITIES_VIDEO_OUTPUT>;
+
+        // @alt supportedvideodisplays
+        // @property
+        // @brief Retrieves VideoOutputs
+        virtual Core::hresult VideoOutputs(VideoOutput& videoOutputs /* @out @bitmask */) const = 0;
+        // @brief Retrieves DefaultResolution against given video Output
+        virtual Core::hresult DefaultResolution(const VideoOutput videoOutput /* @index */, ScreenResolution& defaultResolution /* @out */) const = 0;
+        // @alt supportedresolutions
+        // @brief Retrieves Resolution against given video Output
+        virtual Core::hresult Resolutions(const VideoOutput videoOutput /* @index */, ScreenResolution& resolutions /* @out @bitmask */) const = 0;
+        // @alt supportedhdcp
+        // @brief Retrieves Hdcp
+        virtual Core::hresult Hdcp(const VideoOutput videoOutput /* @index */, CopyProtection& hdcpVersion /* @out */) const = 0;
+        // @property
+        // @brief Retrieves Host EDID
+        virtual Core::hresult HostEDID(string& edid /* @out */) const = 0;
+        // @property
+        // @brief Retrieves HDR
+        virtual Core::hresult HDR(bool& supportsHDR /*@out*/) const = 0;
+        // @property
+        // @brief Retrieves Atoms
+        virtual Core::hresult Atmos(bool& supportsAtmos /*@out*/) const = 0;
+        // @property
+        // @brief Retrieves cec
+        virtual Core::hresult CEC(bool& supportsCEC /*@out*/) const = 0;
+
+        // @brief Retrieves DeviceVideoCapabilities
+        virtual Core::hresult DeviceVideoCapabilities(string& edid /* @out */,
+            bool& hdr /* @out */, bool& atmos /* @out */, bool& cec /* @out */,
+            IVideoOutputCapsIterator*& videoOutputCaps /* @out */) const = 0;
+    };
+
+    /* @json 1.0.0 */
+    struct IDeviceMetadata : virtual Core::IUnknown {
+        enum { ID = ID_DEVICE_METADATA };
+
+        virtual ~IDeviceMetadata() override = default;
+
+        struct Device {
+            string deviceType /* @brief Device type */;
+            string distributorId /* @brief Partner ID or distributor ID for device */;
```

Comment:
Some of these fields seem optional... distributorid, friendlyname, sku... ?

Reply:
done